Pearson-Advance/frontend-app-lti
# feat: add table to list courses & complete deep linking flow

> [!CAUTION]
> To avoid a very long PR I split the effort of solving the ticket PAD-1543 into three. **This is the second one** 
> The PRs are accumulative and need the changes from the previous PR; in their `base` branch they will be pointing to the branch of the previous PR so they show only the changes applied in that specific PR

> [!NOTE]
> Once the previous PR is merged, the base of this branch will change to `main`

### Ticket

PAD-1543

### Description

This PR adds a table to list the institution-licensed courses

### Changes made

- [src/features/Courses/CoursesFilters/index.jsx](https://github.com/Pearson-Advance/frontend-app-lti/pull/5/files#diff-0b1d7f9ae316f053b0f674029f2aa187a023e4d63c3f8626215cb956388162eb): Creates the `Input` to filter by text
- [src/features/Courses/CoursesPage/index.jsx](https://github.com/Pearson-Advance/frontend-app-lti/pull/5/files#diff-8a3d072d2bedf7bbfadb4de6a4b6a6b4e311ed48e8f8c008b7db0fc017ebed41): Unify the components that structure the table and manages the required handlers. Here, the Deep Linking flow is validated to send the selected courses to the requesting platform and is completed by submitting the HTML form obtained through the `validateContentSelection` request
- [src/features/Courses/CoursesTable/index.jsx
](https://github.com/Pearson-Advance/frontend-app-lti/pull/5/files#diff-27ba03c043725a2d55a76c69267144c39c1075b61fce7c48fa2551f55db5a29d): In future, `actionButton` will redirect to the table of classes with the course ID
- [src/index.scss](https://github.com/Pearson-Advance/frontend-app-lti/pull/5/files#diff-6ea08c95a60c10153bc1d5ae1e330e1fa103cc61df85954a91f741359450a5b0) and [src/features/Main/index.scss](https://github.com/Pearson-Advance/frontend-app-lti/pull/5/files#diff-253d0f09fb38c7550eb2add3a12d8ec2de911ba8b06633394cef31a13e3f56aa): One was moved of dir. I have pending to clean up the `SCSS` files

### Screenshots

![image](https://github.com/user-attachments/assets/b26e2b05-a265-41b7-a632-2288a1dbb5e1)

### How to test
- Set the environment to use `Deep Linking`
- Enable CCX in a course and create a Class of that course
- Enable the redirect to the LTI MFE when launching a `Deep Linking Flow`
- Up the LTI MFE 
- 
If everything goes well, once you launch a Deep Linking Flow (I use IMS), you should be redirected to the LTI MFE and you should watch the courses table with the class(es) you created before
